Lib/random.py

```diff
@@ -295,11 +295,10 @@ def _randbelow_without_getrandbits(self, n, int=int, maxsize=1<<BPF):
 
     def choice(self, seq):
         """Choose a random element from a non-empty sequence."""
-        try:
-            i = self._randbelow(len(seq))
-        except ValueError:
-            raise IndexError('Cannot choose from an empty sequence') from None
-        return seq[i]
+        n = len(seq)
+        if n == 0:
+            raise IndexError('Cannot choose from an empty sequence')
+        return seq[self._randbelow(n)]
```

Comment:
It is better to use EAFP. It add less overhead (and may add zero overhead soon).

Reply:
I don't care about overhead here. I would ask you to concentrate on the functionality, not on silly micro-optimizations.

Reply:
However, I will add a check to `_randbelow` itself to ensure that it doesn't enter an infinite loop.

Reply:
The code for *choice()* doesn't need to change at all.   The tests still pass with the original code.

Reply:
You're right, the change isn't necessary anymore now that I've changed `_randbelow` instead.